pkg/controller/kogitodataindex/kogitodataindex_controller.go

```diff
@@ -149,23 +148,8 @@ const (
 	protoBufKeyFolder                  string = "KOGITO_PROTOBUF_FOLDER"
 	protoBufKeyWatch                   string = "KOGITO_PROTOBUF_WATCH"
 	protoBufConfigMapVolumeDefaultMode int32  = 420
-
-	// Collection of kafka topics that should be handled by the Data Index
-	kafkaTopicNameProcessInstances  string = "kogito-processinstances-events"
-	kafkaTopicNameUserTaskInstances string = "kogito-usertaskinstances-events"
-	kafkaTopicNameProcessDomain     string = "kogito-processdomain-events"
-	kafkaTopicNameUserTaskDomain    string = "kogito-usertaskdomain-events"
-	kafkaTopicNameJobsEvents        string = "kogito-jobs-events"
 )
 
-var kafkaTopics = []services.KafkaTopicDefinition{
-	{TopicName: kafkaTopicNameProcessInstances, MessagingType: services.KafkaTopicIncoming},
-	{TopicName: kafkaTopicNameUserTaskInstances, MessagingType: services.KafkaTopicIncoming},
-	{TopicName: kafkaTopicNameProcessDomain, MessagingType: services.KafkaTopicIncoming},
-	{TopicName: kafkaTopicNameUserTaskDomain, MessagingType: services.KafkaTopicIncoming},
-	{TopicName: kafkaTopicNameJobsEvents, MessagingType: services.KafkaTopicIncoming},
-}
-
```

Comment:
only `kafka.bootstrap.servers` property need to be define. User don't need to explicitly define these topics even for data-index

Reply:
Hi! Actually the Topics are required if the cluster is not set to auto create topics. 

Since we do know which topics we require for the supporting services (like Data Index and Trusty), we should create them if KogitoInfra is associated with a Kafka CR.

Reply:
done